iceoryx2-pal/posix/src/macos/mman.rs

```diff
@@ -74,23 +78,94 @@ unsafe fn shm_file_path(name: *const char, suffix: &[u8]) -> [u8; MAX_PATH_LENGT
     state_file_path
 }
 
+unsafe fn get_real_shm_name(name: *const char) -> Option<[u8; SHM_MAX_NAME_LEN]> {
+    let shm_file_path = shm_file_path(name, SHM_STATE_SUFFIX);
+    let shm_state_fd = open_with_mode(shm_file_path.as_ptr().cast(), O_RDONLY, 0);
+    if shm_state_fd == -1 {
+        return None;
+    }
+
+    let mut buffer = [0u8; SHM_MAX_NAME_LEN];
+
+    if read(shm_state_fd, buffer.as_mut_ptr().cast(), SHM_MAX_NAME_LEN) <= 0 {
+        close(shm_state_fd);
+        return None;
+    }
+
+    close(shm_state_fd);
+    Some(buffer)
```

Comment:
Does the buffer need to have a null-termination? I mean, is it possible that read write a non-null character at the last index of the buffer?

Reply:
Yes, since it is used as input argument for `shm_open` which requires a null terminated c string.

The theoretical maximum length of a generated shm name is 32 (but can never be reached), with null terminator it results in `SHM_MAX_NAME_LEN=33`.

I took your note into consideration and read only SHM_MAX_NAME_LEN - 1 characters in read so that the null character is always preserved at the end.